implementations/micrometer-registry-cloudwatch2/src/main/java/io/micrometer/cloudwatch2/CloudWatchMeterRegistry.java

```diff
@@ -295,6 +296,10 @@ StandardUnit toStandardUnit(@Nullable String unit) {
 
         private List<Dimension> toDimensions(List<Tag> tags) {
             return tags.stream()
+                    // Take restrictions for dimensions in CloudWatch into account
+                    // https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_Dimension.html
+                    .filter(t -> StringUtils.isNotBlank(t.getValue()))
```

Comment:
If we are dropping a tag, we could use our `WarnThenDebugLogger` to give some visibility to users that a tag is being dropped and for what reason. Otherwise, it may be too hard to notice and track down for users.

Reply:
The implementation feels a bit clumsy but since we're operating on a stream I did not have a better idea.